src/lib/containers/markdown/MarkdownWidget.tsx

```diff
@@ -1,15 +1,74 @@
 import React from 'react'
-import MarkdownButton from './widget/MarkdownButton'
-import MarkdownIDUReport from './widget/MarkdownIDUReport'
-import MarkdownSynapseImage from './widget/MarkdownSynapseImage'
-import MarkdownSynapsePlot from './widget/MarkdownSynapsePlot'
+import MarkdownButton, { ButtonLinkWidgetParams } from './widget/MarkdownButton'
+import MarkdownIDUReport, {
+  MarkdownIDUReportProps,
+} from './widget/MarkdownIDUReport'
+import MarkdownSynapseImage, {
+  ImageWidgetParams,
+} from './widget/MarkdownSynapseImage'
+import MarkdownSynapsePlot, {
+  PlotWidgetParams,
+} from './widget/MarkdownSynapsePlot'
+import MarkdownSynapseTable, {
+  MarkdownSynapseTableProps,
+} from './widget/MarkdownSynapseTable'
 import MarkdownTableOfContents from './widget/MarkdownTableOfContents'
-import MarkdownUserBadge from './widget/MarkdownUserBadge'
-import MarkdownVideo from './widget/MarkdownVideo'
+import MarkdownUserBadge, {
+  MarkdownUserBadgeProps,
+} from './widget/MarkdownUserBadge'
+import MarkdownVideo, { MarkdownVideoProps } from './widget/MarkdownVideo'
 
-export type MarkdownWidgetProps = {
-  widgetType: string
-  widgetParamsMapped: any
+type ButtonLink = {
```

Comment:
Trying to accomplish improved type safety using the same pattern we use for `SynapseConfig` in portals